pkg/fake/atomic.go

```diff
@@ -159,3 +159,11 @@ func (a *AtomicPtrSlice[T]) Pop() *T {
 	a.values = a.values[0 : len(a.values)-1]
 	return last
 }
+
+func (a *AtomicPtrSlice[T]) ForEach(fn func(*T)) {
+	a.mu.Lock()
```

Comment:
nit: Doesn't this just need a ReadLock since you aren't actually writing against anything that's being guarded?